bsmetadata/preprocessing_utils.py

```diff
@@ -591,3 +622,86 @@ def preprocess(self, examples: Dict[str, List]) -> Dict[str, List]:
 
         examples[self.col_to_store_metadata] = example_metadata_list
         return examples
+
+
+class ErrorWrapperPreprocessor:
```

Comment:
This class is useful for mass extraction: if I have an example that returns an error it does not prevent the processing of other examples and it allows me to "know" the type of error returned on this particular example (what we do not see otherwise)